compiler/rustc_typeck/src/check/coercion.rs

```diff
@@ -1500,7 +1500,8 @@ impl<'tcx, 'exprs, E: AsCoercionSite> CoerceMany<'tcx, 'exprs, E> {
                             coercion_error.clone(),
                             fcx,
                             parent_id,
-                            expression.map(|expr| (expr, blk_id)),
+                            expression,
+                            Some(blk_id),
```

Comment:
Technically, this is not the same as previously: if `expression` is `None` previously it will be `None` and now `Some`. Changing that is as simple as `expression.and(Some(blk_id))`, but I don't think this is necessary.